### PR TITLE
feat: bulk artist identification engine with 3-tier pipeline

### LINKED ISSUE #553: Bulk artist identification engine with album-comparison scoring
[mode: plan]
[model: opus]
[effort: high]

## Summary

Add a bulk artist identification engine that searches configured providers and uses album-directory comparison to automatically link unidentified artists to MusicBrainz records.

## Problem / Motivation

In a greenfield scenario (well-populated folder structure, no NFO files), the filesystem scanner creates skeleton artist records with only a name and path. Users must then manually re-identify each artist one-by-one via the disambiguation UI. For collections with hundreds of artists, this is impractical.

The Re-Identify Artist feature already has all the building blocks (provider search, album comparison, confidence scoring), but they only work for a single artist at a time.

## Proposed Solution

A bulk identification service that processes a list of unidentified artists through a 3-tier pipeline:

### Tier 1 -- Connection-based identification (highest confidence)
If Emby/Jellyfin/Lidarr connections exist and have been scanned, use their MBID mappings. Match connection-library artists to filesystem-library artists by MBID or case-insensitive name and propagate the provider ID. Confidence: ~1.0.

### Tier 2 -- Album-comparison identification (high confidence)
For artists not matched by connections, search MusicBrainz by folder name. For the top results, use `ListLocalAlbums()` + `CompareAlbums()` to score each candidate against album subdirectory names. Rate-limiter-aware (MusicBrainz 1 req/sec).

### Tier 3 -- Name-only search (lower confidence)
For artists with no album subdirectories, fall back to name-only MusicBrainz search.

### Confidence-based categorization

Results are categorized into three buckets:
- **Auto-link**: MBID from connection (tier 1), or album match >= 70% with exactly 1 strong candidate (tier 2). These are linked without user intervention.
- **Review queue**: Multiple tier-2 candidates above 30%, or single candidate between 30-70%. Presented to user for selection.
- **Unmatched**: No candidates above 30%, or zero album data. Flagged for manual re-identification.

### Reusable components

| Component | Location | Reuse |
|-----------|----------|-------|
| `SearchForLinking(query, providers)` | `provider/orchestrator.go` | Search MB/Discogs by name |
| `ListLocalAlbums(path)` | `artist/albums.go` | Read album folder names |
| `CompareAlbums(local, remote)` | `artist/albums.go` | Score match percentage |
| `ReleaseGroupFetcher.GetReleaseGroups(mbid)` | `provider/provider.go` | Get remote album titles |
| `enrichWithAlbumComparison()` | `api/handlers_refresh.go` | Wire up comparison |
| `handleRefreshLink()` flow | `api/handlers_refresh.go` | Link artist + fetch metadata |

### Progress tracking

Uses an in-memory progress tracker (similar to fix-all) with mutex protection. Reports: total artists, processed count, auto-linked count, review-needed count, unmatched count, current artist name.

### API

- `POST /api/v1/artists/bulk-identify` -- start bulk identification (accepts optional library_id filter)
- `GET /api/v1/artists/bulk-identify/progress` -- poll progress
- `DELETE /api/v1/artists/bulk-identify` -- cancel in-progress run
- Only one bulk identification can run at a time (409 on concurrent starts)

## Acceptance Criteria

- [ ] Bulk identification engine processes unidentified artists through the 3-tier pipeline
- [ ] Connection-based identification uses cross-library MBID/name matching (builds on #465)
- [ ] Album-comparison scoring reuses existing `CompareAlbums()` logic
- [ ] Rate limiting respected for MusicBrainz API calls
- [ ] Results categorized into auto-link / review / unmatched buckets
- [ ] Auto-linked artists have full metadata fetched (MBID, biography, provider IDs)
- [ ] Progress tracking with cancel support
- [ ] API endpoints for start, progress, and cancel
- [ ] Unit tests for the identification pipeline and confidence categorization

## Additional Context

- Depends on #465 (cross-library platform ID propagation) for tier-1 connection-based matching
- The Re-Identify feature (`handlers_refresh.go`) already proves the album-comparison approach works for individual artists
- The confidence thresholds (70% auto-link, 30% review) match the existing `albumMatchClasses()` color coding in the disambiguation UI
- Related: #554 (OOBE integration that will consume this engine)

### PR DESCRIPTION
**Pull request by sydlexius**
## Summary

- Add bulk identification engine that processes unidentified artists through a 3-tier pipeline: connection-based matching (Tier 1), album-comparison scoring (Tier 2), and name-only search (Tier 3)
- Auto-links high-confidence matches (connection MBID, >=70% album match, single high-score name result) and queues medium-confidence matches for manual review
- Extract `executeRefreshCtx` from `executeRefresh` for reuse from background goroutines

Closes #553

## Test plan

- [ ] `go test ./internal/api/... -run TestBulkIdentify` -- 14 new tests covering singleton 409 rejection, progress polling, cancel, tier 1 connection match, excluded/locked skip, link endpoint
- [ ] `go test ./... -count=1` -- no regressions (full suite passes)
- [ ] Manual: POST `/api/v1/artists/bulk-identify`, poll progress via GET, cancel via DELETE
- [ ] Manual: verify review queue populates for ambiguous matches, link via POST `.../link`

🤖 Generated with [Claude Code](https://claude.com/claude-code)

## Summary by CodeRabbit

* **New Features**
  * Bulk artist identification: start, monitor (detailed progress & review queue), and cancel jobs (concurrent starts rejected)
  * Manual linking from the review queue to set MusicBrainz/Discogs IDs and refresh metadata
  * Multi‑tier matching with auto‑linking, queued review candidates, and optional library scoping

* **Documentation**
  * OpenAPI schemas and endpoints for bulk-identify operations and linking

* **Tests**
  * End-to-end tests covering endpoints, lifecycle, matching tiers, and edge cases

### CODE REVIEW

File: internal/api/handlers_identify.go [hunk 1]
```diff
@@ -0,0 +1,688 @@
...
+				if r.orchestrator != nil {
+					if _, err := r.executeRefreshCtx(ctx, a); err != nil {
+						r.logger.Warn("bulk-identify: Tier 1 refresh failed",
+							"artist", a.Name, "error", err)
+						// Still count as auto-linked since MBID was set.
+					}
+				}
+				rule.EvaluateAndPersistHealth(ctx, r.ruleEngine, r.artistService, a, r.logger)
+				return identifyResult{Outcome: outcomeAutoLinked}
+			}
```
[the remark this answers is not in the document]
Comment (sydlexius) [reply internal/api/handlers_identify.go]: Fixed in b26a999.

File: internal/api/openapi.yaml [hunk 2]
```diff
@@ -2824,6 +2934,150 @@ paths:
...
+                properties:
+                  status:
+                    type: string
+                    example: canceled
+                  message:
+                    type: string
+                    example: bulk identification canceled
+                required:
+                  - status
+                  - message
```
[the remark this answers is not in the document]
Comment (sydlexius) [reply internal/api/openapi.yaml]: Fixed in b26a999.

**Comment by sydlexius** [comment]
@coderabbitai resolve
